### PR TITLE
feat: Add simple opentelemetry-collector config for use in dev

### LINKED ISSUE #74: chore: OTel collector for exploratory testing
Reviewing and and performing exploratory testing PRs would be made easier by being able to visualize the results in a tracing backend. 

This repository could provide contributors with an OTel collector service configured in the [main `docker-compose` file](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/docker-compose.yml) like we do for other services.  

Inspired by: https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/72#issuecomment-1170077416

### PR DESCRIPTION
**Pull request by ahayworth**
Mostly this is borrowed from open-telemetry/opentelemetry-demo-webstore.

This commit adds the opentelemetry-collector and the jaeger all-in-one
image to the docker-compose file. This allows us to export traces in
development to a collector and then visualize them in the Jaeger UI.

Fixes #74

### CODE REVIEW

**Review by arielvalentin — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -109,6 +109,15 @@ To test using Docker:
      2. Run the tests for the sdk
          *  `docker-compose run sdk bundle exec rake test`
 
+## OpenTelemetry Collector
```
Comment (plantfansam) [inline CONTRIBUTING.md:112]:
```suggestion
## Processing and visualizing traces locally
```

File: otelcol-config.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+receivers:
+  otlp:
+    protocols:
+      grpc:
+      http:
+exporters:
+  jaeger:
+    endpoint: "jaeger:14250"
```
Comment (plantfansam) [inline otelcol-config.yml:8]: Is this the correct point? I think I see Jaeger listening on `16686` above, but I haven't worked much in this area...

**Review by plantfansam — APPROVED** [review]
Couple nits, but I'm sure I'm missing something. Happy to merge and sort out any confusion in follow-on PRs, as this is just for dev.

File: otelcol-config.yml (hunk 2 above)
Comment (ahayworth) [reply otelcol-config.yml:8]: 16686 is the jaeger UI port - this port is for the collector to export to the jaeger storage backend itself.

Comment (ahayworth) [reply otelcol-config.yml:8]: Oh, also I suppose I should note that the explicit mention of 16686 up there doesn't mean that jaeger is only listening on that port - it's a docker-compose directive saying "please map port 16686 from the container network into the host network." :)

**Pull request merged**